packages/web-client/app/models/workflow.ts

```diff
@@ -51,10 +50,29 @@ export abstract class Workflow {
   workflowPersistence: WorkflowPersistence;
   workflowPersistenceId?: string;
 
-  constructor(owner?: any) {
+  abstract restorationErrors(state: any): string[];
```

Comment:
My reasoning was that since the concrete workflows need to implement validation and restoration checks, `abstract` makes sense here. However, this comes with extra burden of needing to implement these methods on every instantiation, which is reflected in the later commit where I introduce `WorkflowStub` that is used in unit and integration tests.